### DISCUSSION TITLE
face swap

### DISCUSSION THREAD

**Discussion by sheriffdude** [runew0lf/RuinedFooocus#104] [Ideas]
can you add face swap //like roop or reactor that would be great ..well done guys great program

**Comment by yownas** [comment]
We actually used to have faceswapping. But it had to be removed beacuse of two major reasons.

First, the onnxruntime module for python caused too many issues where it wouldn't install properly for some users.

But the main reason is that we don't  feel comfortable using code/models from Insighface. There were some big drama about them not liking how their models were being used and they sent a bunch of dmca-takedowns to people. 

So, even if their model wasn't a crappy 128x128 pixels, there is no way I'd go near anything they made. Sorry.

**Comment by heathcliff2019** [comment]
Hey @yownas , since faceswap is not incl. anymore can you explain how I can use one of the same model for future protects? ie. I want to use one of the same consistent character.

**Comment by yownas** [comment]
The best alternative (that I know of) is FaceFusion https://github.com/facefusion/facefusion

But if you _really_ want faceswapping in RuinedFooocus you can enable it manually by running the script  `RuinedFooocus\tools\enable_faceswap.bat` . It does it best to install the libraries needed, but it might also just fail and force you to reinstall. So I won't  recommend it.
